layouts/partials/header.html

```diff
@@ -33,14 +33,14 @@
 			<div id="navbar" class="navbar-collapse collapse">
 				<ul class="nav navbar-nav">
 					<li {{ if eq $isHomePage true }}class="active"{{ end }}><a href="{{ $baseUrl }}/">Home</a></li>
-			{{ if isset .Site.Params "mainMenu" }}
+			{{ if .Site.Params.mainMenu }}
 				{{ $url := .Permalink }}
 				{{ range $menu := .Site.Params.mainMenu }}
```

Comment:
I think if you use `with` above, this becomes `{{ range $menu := . }}`.

The Hugo documentation says, "`with` rebinds the context `.` within its scope", so this should work.